portfolio/src/main/java/com/google/sps/servlets/RandomStateServlet.java

```diff
@@ -31,17 +36,26 @@
   private Gson gson = new Gson();
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Map<String, Double> njCoords = new HashMap<>();
-    njCoords.put("lat", 40.22);
-    njCoords.put("lng", -74.76);
-    stateCapitalCoords.put("New Jersey", njCoords);
-    Map<String, Double> nyCoords = new HashMap<>();
-    nyCoords.put("lat", 42.65);
-    nyCoords.put("lng", -73.76);
-    stateCapitalCoords.put("New York", nyCoords);
-    
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    readStateCoords();
     response.setContentType("application/json");
     response.getWriter().println(gson.toJson(stateCapitalCoords));
   }
+
+  private void readStateCoords() throws IOException, ServletException {
+    InputStream stateCoords = getServletContext().getResourceAsStream("/WEB-INF/state_capitals.txt");
+    InputStreamReader inputReader = new InputStreamReader(stateCoords);
+    BufferedReader reader = new BufferedReader(inputReader);
+    String data;
+    while ((data = reader.readLine()) != null) {
+      String[] dataSegments = data.split(" +");
+      String stateName = dataSegments[0];
+      double lat = Double.parseDouble(dataSegments[1]);
+      double lng = Double.parseDouble(dataSegments[2]);
```

Comment:
Do I still want to log possible errors from parseDouble here even though it is just reading from a file where we know the input?